### PR TITLE
style(Tag): Update Tag colors

### PR DESCRIPTION
**Pull request by LinKCoding**
### Overview
Updates tags to use the correct color tokens.
Current implementation of tags somewhat uses ColorMode and needs to invert the colors to work. 

Major Changes:
1. removed DismissButton from TagWrapper
2. TagWrapper => TagLabelWrapper 
3. ^ necessary to remove the background behind DimissButton 
4. Tag is no longer a Background, is now a Box 
5. Changed the BaseButton to an IconButton (to gain access to ToolTip properties) 
6. The gray variant should have updated colors (and for hover state too)

Updates tags to use the correct color tokens.

### PR Checklist

- [ ] Related to designs: https://www.figma.com/design/ReGfRNillGABAj5SlITalN/%F0%9F%93%90-Gamut?node-id=46539-40954&m=dev 
- [x] Related to JIRA ticket: [GM-778]
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change
- [x] This PR includes testing instructions tests for the code change
- [x] The alpha package of this PR is passing end-to-end tests in all relevant Codecademy repositories

#### Testing Instructions

1. Go to the Storybook preview and the tags page `?path=/docs/atoms-tag--tag-variants` 
2. Look over the tags implementation, compare against the [figma](https://www.figma.com/design/ReGfRNillGABAj5SlITalN/%F0%9F%93%90-Gamut?node-id=46539-40954&m=dev) 
—
Mono
1. Go to the tengu preview env
2. Go to Pricing > Individuals > select "Try Pro for free" 
3. Create an account
4. At checkout, click on the "Have a promo code?" link
5. Add in `test50` 
6. Check that the tag renders correctly. 
— 
Monolith
1. go to the preview (without portal app), log in as an admin
2. Go to the admin page and create a test user that has a Pro subscription + billed monthly (you can also use this link: 
https://pr-38096-monolith.dev-eks.codecademy.com/admin/test_user?prefill_id=66ab9a074e03aa000159fe35)
^ the settings is important because we need to upgrade the user (and can't seem to upgrade free or pro-lite users) 
3. Impersonate the test user 
4. go to "Account + Billing" > "Payment And Plan" tab 
5. Select "Change Plan" and opt to upgrade to bill yearly 
6. in the address bar append the query string: `?coupon=test50` 
7. see that the tag is rendered 
8. ... 
9. Profit 

Screenshots:
Mono:
image

Monolith: 
image

 

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/38096)  | [Monolith Env](http://www.google.fr/ 'Named link title') |
| Portal       | [Portal Link](https://github.com/codecademy-engineering/mono/pull/7067)  | [Portal Env](http://www.google.fr/ 'Named link title')   |
| Another Repo | [Another Link](http://www.google.fr/ 'Named link title') | [Another Env](http://www.google.fr/ 'Named link title')  |

[GM-778]: https://codecademy.atlassian.net/browse/GM-778?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.25.4-alpha.cbf226.0<br/>@codecademy/gamut-kit@0.6.425-alpha.cbf226.0<br/>@codecademy/styleguide@66.29.5-alpha.cbf226.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://669b274c09fafcf572478830--gamut-preview.netlify.app>https://669b274c09fafcf572478830--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/669b274c09fafcf572478830>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.25.4-alpha.55c54a.0<br/>@codecademy/gamut-kit@0.6.425-alpha.55c54a.0<br/>@codecademy/styleguide@66.29.5-alpha.55c54a.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@55.25.4-alpha.6ee5ff.0<br/>@codecademy/gamut-kit@0.6.425-alpha.6ee5ff.0<br/>@codecademy/styleguide@66.29.5-alpha.6ee5ff.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://669f17250c86fb93269548cf--gamut-preview.netlify.app>https://669f17250c86fb93269548cf--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/669f17250c86fb93269548cf>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@56.0.3-alpha.7c860c.0<br/>@codecademy/gamut-kit@0.6.428-alpha.7c860c.0<br/>@codecademy/styleguide@67.0.3-alpha.7c860c.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://669fba26052f962d4dad37fc--gamut-preview.netlify.app>https://669fba26052f962d4dad37fc--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/669fba26052f962d4dad37fc>Deploy Logs</a></p>

**Comment by dreamwasp** [comment]
image
 
this looks great, i don't see the line on zoom at all so i think it's just a weird pixel thing 🤷  only issue i'm seeing is this strange focus state bug

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@56.0.3-alpha.eb7fd3.0<br/>@codecademy/gamut-kit@0.6.428-alpha.eb7fd3.0<br/>@codecademy/styleguide@67.0.3-alpha.eb7fd3.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://669fc5313774f94c1ddf9a22--gamut-preview.netlify.app>https://669fc5313774f94c1ddf9a22--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/669fc5313774f94c1ddf9a22>Deploy Logs</a></p>

**Comment by LinKCoding** [comment]
> this looks great, i don't see the line on zoom at all so i think it's just a weird pixel thing 🤷 only issue i'm seeing is this strange focus state bug (prob some inherited style 🤔 )

Nice, thanks! - I realized that focus states are a little messed up after I implemented my changes, gonna take some time to fix that 
(Just need to do the truncated tag for some reason the outline doesn't reach the full component)

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@56.0.3-alpha.9dca7d.0<br/>@codecademy/gamut-kit@0.6.428-alpha.9dca7d.0<br/>@codecademy/styleguide@67.0.3-alpha.9dca7d.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://66a2bb82bc7eeeb858887328--gamut-preview.netlify.app>https://66a2bb82bc7eeeb858887328--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/66a2bb82bc7eeeb858887328>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@56.0.3-alpha.e96907.0<br/>@codecademy/gamut-kit@0.6.428-alpha.e96907.0<br/>@codecademy/styleguide@67.0.3-alpha.e96907.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://66a2bee258937ead4c286b6b--gamut-preview.netlify.app>https://66a2bee258937ead4c286b6b--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/66a2bee258937ead4c286b6b>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@56.0.3-alpha.fdf94d.0<br/>@codecademy/gamut-kit@0.6.428-alpha.fdf94d.0<br/>@codecademy/styleguide@67.0.3-alpha.fdf94d.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://66a3b86b223e6b155f48b128--gamut-preview.netlify.app>https://66a3b86b223e6b155f48b128--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/66a3b86b223e6b155f48b128>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@56.0.3-alpha.08d171.0<br/>@codecademy/gamut-kit@0.6.428-alpha.08d171.0<br/>@codecademy/styleguide@67.0.3-alpha.08d171.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://66a3bb43d67327009c661828--gamut-preview.netlify.app>https://66a3bb43d67327009c661828--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/66a3bb43d67327009c661828>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://66a3be781fc25e00a9b3502b--gamut-preview.netlify.app>https://66a3be781fc25e00a9b3502b--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/66a3be781fc25e00a9b3502b>Deploy Logs</a></p>

**Review by dreamwasp — COMMENTED** [review]
these look great! only thing i'm noticing is something slightly off with the focus styles on hover of one of the examples - the focus styles seem to be taking up the full width of their container? 

image

if this is only happening in this example and not in actual code i'm happy to ✅  so i'm gonna take a look at mono now~

File: packages/gamut/src/Tag/elements.tsx [hunk 1]
```diff
@@ -23,10 +24,9 @@ export const tagProps = variance.compose(
 export const Outline = styled(Box)(
   css({
     // this is a bit of a hack as we don't have access to focus-visible from this component.  if you are not properly dismissing your tags you may see this primary colored outline after clicking X, but otherwise you should never hit this behavior.
-
     borderRadius: tagBorderRadius,
-    width: 'fit-content',
-    maxWidth: '100%',
+    width: '100%',
+    maxWidth: 'fit-content',
```
[the remark this answers is not in the document]
Comment (LinKCoding) [reply packages/gamut/src/Tag/elements.tsx:29]: lol this was your code! 😂

**Comment by LinKCoding** [comment]
> these look great! only thing i'm noticing is something slightly off with the focus styles on hover of one of the examples - the focus styles seem to be taking up the full width of their container?
> 
> image
> 
> if this is only happening in this example and not in actual code i'm happy to ✅ so i'm gonna take a look at mono now~

I see the issue — I'm guessing to add more spacing around tags, there was a .5 rem margin added around the tag itself which makes the focus outline bigger. 

I've added a box around the tag now to include this padding and this should fix the issue. 

Thanks for catching!!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@56.0.5-alpha.5b8a25.0<br/>@codecademy/gamut-kit@0.6.430-alpha.5b8a25.0<br/>@codecademy/styleguide@67.0.5-alpha.5b8a25.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://66a7a98b55a8ef16d50294dd--gamut-preview.netlify.app>https://66a7a98b55a8ef16d50294dd--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/66a7a98b55a8ef16d50294dd>Deploy Logs</a></p>

File: packages/gamut/src/Tag/elements.tsx [hunk 2]
```diff
@@ -37,38 +37,44 @@ export const Outline = styled(Box)(
...
-      borderRadiusRight: tagBorderRadius,
-      color: 'currentColor',
+      ...iconButtonOverrides,
+      color: 'background',
+      border: 'none',
+      borderRadius: `0 ${tagBorderRadius} ${tagBorderRadius} 0`,
+      width: 12,
+      [hoverAndFocus]: {
+        color: 'background',
+        bg: 'secondary-hover',
```
Comment (dreamwasp) [inline packages/gamut/src/Tag/elements.tsx:61]: 🔥

File: packages/gamut/src/Tag/styles.tsx [hunk 3]
```diff
@@ -1,47 +1,58 @@
...
   alignItems: 'center',
-  justifyContent: 'center',
-  borderRadius: tagBorderRadius,
+  borderRadius: `${tagBorderRadius} 0 0 ${tagBorderRadius}`,
+  display: 'flex',
   height: '24px',
-  width: 'fit-content',
+  justifyContent: 'center',
   maxWidth: '100%',
+  // width: 'fit-content',
```
Comment (dreamwasp) [inline packages/gamut/src/Tag/styles.tsx:16]: commented out code?

**Review by dreamwasp — COMMENTED** [review]
this is super weird but it looks like the remove buttons on multiselect tags lost their right border radius somehow 🤔 
image

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@56.0.5-alpha.f5a356.0<br/>@codecademy/gamut-kit@0.6.430-alpha.f5a356.0<br/>@codecademy/styleguide@67.0.5-alpha.f5a356.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://66a90092c2b33e34d5b96552--gamut-preview.netlify.app>https://66a90092c2b33e34d5b96552--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/66a90092c2b33e34d5b96552>Deploy Logs</a></p>

**Comment by dreamwasp** [comment]
The SelectDropdown multiselect tag border radius are still a _little_ off (2px v 4px in tags)- i think `borderRadius` got moved out of `tagBaseStyles` or something in the `multiValue` object so it's not making its way there anymore. you should be able to fix it by adding borderRadius 4px [here] (https://github.com/Codecademy/gamut/blob/f5a35661fe1f7262dc42effe6c84df3b240aff37/packages/gamut/src/Form/SelectDropdown/styles.ts#L177) 
image

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@56.0.5-alpha.400a71.0<br/>@codecademy/gamut-kit@0.6.430-alpha.400a71.0<br/>@codecademy/styleguide@67.0.5-alpha.400a71.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://66aa43bf91a02e007768e59b--gamut-preview.netlify.app>https://66aa43bf91a02e007768e59b--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/66aa43bf91a02e007768e59b>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@56.0.6-alpha.3cb783.0<br/>@codecademy/gamut-kit@0.6.431-alpha.3cb783.0<br/>@codecademy/styleguide@67.0.6-alpha.3cb783.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://66aa9da219edfd043ad405c5--gamut-preview.netlify.app>https://66aa9da219edfd043ad405c5--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/66aa9da219edfd043ad405c5>Deploy Logs</a></p>

File: packages/gamut/src/Form/SelectDropdown/styles.ts [hunk 4]
```diff
@@ -174,6 +174,7 @@ export const getMemoizedStyles = (
       ...tagBaseStyles,
       cursor: 'pointer',
       background: theme.colors.background,
+      borderRadius: '4px',
```
Comment (dreamwasp) [inline packages/gamut/src/Form/SelectDropdown/styles.ts:177]: you could grab this from the tagBorderRadius var just so we're ensuring it stays consistent!

**Review by dreamwasp — COMMENTED** [review]
not sure when but at some point in the last couple commits the default button styling came back on focus 🤔 

image

**Review by dreamwasp — APPROVED** [review]
these look wonderful! beautiful job ✨

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@56.0.6-alpha.69b494.0<br/>@codecademy/gamut-kit@0.6.431-alpha.69b494.0<br/>@codecademy/styleguide@67.0.6-alpha.69b494.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://66ab9349574f3d31456543ce--gamut-preview.netlify.app>https://66ab9349574f3d31456543ce--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/66ab9349574f3d31456543ce>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://66ad3dd0a22e2abecaea36dd--gamut-preview.netlify.app>https://66ad3dd0a22e2abecaea36dd--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/66ad3dd0a22e2abecaea36dd>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut@56.1.1-alpha.4f5640.0<br/>@codecademy/gamut-kit@0.6.432-alpha.4f5640.0<br/>@codecademy/styleguide@67.1.1-alpha.4f5640.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://66b0f5496e8dc3241d410074--gamut-preview.netlify.app>https://66b0f5496e8dc3241d410074--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/66b0f5496e8dc3241d410074>Deploy Logs</a></p>

**Pull request merged**